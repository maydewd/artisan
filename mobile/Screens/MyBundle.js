/**
 * MyBundle
 */

import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  ListView,
  Image,
  AsyncStorage,
  TouchableOpacity
} from 'react-native';
var NavigationBar = require('react-native-navbar');
import {getScreenHeight, topNavBarHeight, getScreenWidth} from '../helpers/dimension'
import { SwipeListView } from 'react-native-swipe-list-view';
styles = require('../Styles/Layouts');

class MyBundle extends Component {

  constructor(props) {
    super(props);
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    // TODO: hook up data source to server
    this.state = {
      dataSource: ds.cloneWithRows([]),
    };
  }

  componentDidMount() {
    this._fetchData()
  }

  _fetchData() {
    AsyncStorage.getItem('jwtToken', (err, result) => {
      fetch("http://colab-sbx-137.oit.duke.edu:3000/api/listings/liked",
        {method: "GET",
          headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json',
            'Authorization': result
          }})
        .then((response) => response.json())
        .then((responseData) => {
          console.log(responseData);

            var posts = [];
            responseData.forEach((item) => {
              posts.push(item)
            })
            this.setState({
                dataSource: this.state.dataSource.cloneWithRows(posts)
            });
            console.log("Successfully grabbed data");
         })
        .catch(function(err) {
          alert("error");
          console.log("Error in Posting");
          console.log(err);
        })
        .done();
    });
  }

  render() {
    var titleConfig = {
     title: 'MyBundle',
   };

   const leftButtonConfig = {
    title: 'Back',
    handler: () => {this.pop()}
  };

    return (
      <View>
        <NavigationBar
        style={styles.navBar}
        title={titleConfig}
        leftButton={leftButtonConfig}
        />
        <SwipeListView
          style = {{height: getScreenHeight() -topNavBarHeight()-20}}
          dataSource = {this.state.dataSource}
          renderRow = {this.renderPost}
          renderHiddenRow = {(item) => (
              <TouchableOpacity
              onPress={() => this._delete(item)}
              style = {{height: 80, flexDirection: 'row', justifyContent: 'flex-end', alignItems: 'center', backgroundColor: 'red', width: getScreenWidth()}}>
                <Text>Remove</Text>
              </TouchableOpacity>
            )}
          rightOpenValue={-80}
        />
      </View>
    );
  }

  renderPost(item) {
    return (
        <View
        style = {{flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', width: getScreenWidth(), backgroundColor: 'white',}}>
          <Image style = {{height: 80, width: 80}}
           source = {{uri: "http://colab-sbx-137.oit.duke.edu:3000/" + item.imagePath}}/>
           <Text> End</Text>
         </View>
    )
  }

  _delete(data) {
    console.log("want to delete" + data._id)
  }

  pop() {
    this.props.navigator.pop()
  }
}

module.exports = MyBundle
