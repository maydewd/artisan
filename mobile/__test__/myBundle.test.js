import 'react-native';
import React from 'react';
import MyBundle from '../Screens/MyBundle.js';

import renderer from 'react-test-renderer';

it('MyBundle post renders correctly', () => {
  const tree = renderer.create(
    <MyBundle />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
