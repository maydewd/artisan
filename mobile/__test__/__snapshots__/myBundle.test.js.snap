exports[`test MyBundle post renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
        },
        null,
      ]
    }>
    <View
      style={
        Array [
          Object {
            "height": 20,
          },
          null,
        ]
      } />
    <View
      style={
        Array [
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "space-between",
          },
          Object {
            "backgroundColor": "#cce5ff",
            "height": 45,
          },
        ]
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#333",
                "fontSize": 17,
                "fontWeight": "500",
                "letterSpacing": 0.5,
              },
              null,
            ]
          }>
          MyBundle
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                undefined,
                Object {
                  "marginLeft": 8,
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                    "letterSpacing": 0.5,
                  },
                  Object {
                    "color": "#0076FF",
                  },
                ]
              }>
              Back
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}>
          <View
            style={
              Array [
                undefined,
                Object {
                  "marginRight": 8,
                },
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                    "letterSpacing": 0.5,
                  },
                  Object {
                    "color": "#0076FF",
                  },
                ]
              }>
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <ScrollView
    closeOnRowPress={true}
    closeOnScroll={true}
    dataSource={
      ListViewDataSource {
        "items": 0,
      }
    }
    disableLeftSwipe={false}
    disableRightSwipe={true}
    enableEmptySections={true}
    initialListSize={10}
    leftOpenValue={0}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    previewFirstRow={false}
    recalculateHiddenLayout={false}
    removeClippedSubviews={true}
    renderHiddenRow={[Function]}
    renderRow={[Function]}
    rightOpenValue={-80}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": 602,
      }
    } />
</View>
`;
